base/reflection.jl

```diff
@@ -503,10 +503,13 @@ function _dump_function(f::ANY, t::ANY, native::Bool, wrapper::Bool, strip_ir_me
     meth = func_for_method_checked(meth, tt)
     linfo = ccall(:jl_specializations_get_linfo, Ref{Core.MethodInstance}, (Any, Any, Any), meth, tt, env)
     # get the code for it
-    return _dump_function(linfo, native, wrapper, strip_ir_metadata, dump_module)
+    return _dump_function(linfo, native, wrapper, strip_ir_metadata, dump_module, asmvariant)
 end
 
-function _dump_function(linfo::Core.MethodInstance, native::Bool, wrapper::Bool, strip_ir_metadata::Bool, dump_module::Bool)
+function _dump_function(linfo::Core.MethodInstance, native::Bool, wrapper::Bool, strip_ir_metadata::Bool, dump_module::Bool, asmvariant::Symbol=:att)
+    if asmvariant != :att && asmvariant != :intel
+       throw(ErrorException("asmvariant must be either :intel or :att"))
```

Comment:
this should maybe be `ArgumentError` unless anyone else has a better suggestion for except type
(the indent is also slightly off)